### DISCUSSION TITLE
What is a border vertex?

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7245] [Q&A] [answered]
That's surely explained in the doc but I didn't find. CGAL finds one border vertex in the curved cylinder at left of the figure below. I thought a border vertex is a vertex on a border edge but there's no border edge here, unless there's an issue in this mesh.

![Capture du 2023-02-03 13-04-44](https://user-images.githubusercontent.com/4466543/216599934-1e8371e2-f8ae-4fba-b5ad-4c0cbe569499.png)

**Comment by MaelRL** [comment]
Are you sure there is no border edge or is it visually that it does not look like there is a border edge? Can you please provide the mesh & code?

**Comment by stla** [comment]
Here are the code and the mesh.

```cpp
    std::set<Mesh3::Vertex_index> constrained_vertices;
    for(Mesh3::Vertex_index v : smesh.vertices()) {
      if(smesh.is_border(v)) {
        constrained_vertices.insert(v);
      }
    }
    {
      const size_t nbv = constrained_vertices.size();
      std::string tail;
      if(nbv > 1) {
        tail = std::to_string(nbv) + " border vertices.\n";
      } else {
        tail = " one border vertex.\n";
      }
      Message("Constraining: " + tail);
    }
```

Then `"one border vertex"` is printed.

[curvedCylinder.off.txt](https://github.com/CGAL/cgal/files/10578792/curvedCylinder.off.txt)

**Comment by stla** [comment, marked as answer]
Ahah stupid! Just seen the mistake, sorry!
